pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -177,28 +207,51 @@ func NewCodec(codec runtime.Codec, kind string) runtime.Codec {
 	return &thirdPartyResourceDataCodec{codec, kind}
 }
 
-func (t *thirdPartyResourceDataCodec) populate(objIn *extensions.ThirdPartyResourceData, data []byte) error {
+func parseObject(data []byte) (map[string]interface{}, error) {
 	var obj interface{}
 	if err := json.Unmarshal(data, &obj); err != nil {
 		fmt.Printf("Invalid JSON:\n%s\n", string(data))
-		return err
+		return nil, err
 	}
 	mapObj, ok := obj.(map[string]interface{})
 	if !ok {
-		return fmt.Errorf("unexpected object: %#v", obj)
+		return nil, fmt.Errorf("unexpected object: %#v", obj)
 	}
-	return t.populateFromObject(objIn, mapObj, data)
+	return mapObj, nil
 }
 
-func (t *thirdPartyResourceDataCodec) populateFromObject(objIn *extensions.ThirdPartyResourceData, mapObj map[string]interface{}, data []byte) error {
+func (t *thirdPartyResourceDataCodec) populate(data []byte) (runtime.Object, error) {
+	mapObj, err := parseObject(data)
+	if err != nil {
+		return nil, err
+	}
+	return t.populateFromObject(mapObj, data)
+}
+
+func (t *thirdPartyResourceDataCodec) populateFromObject(mapObj map[string]interface{}, data []byte) (runtime.Object, error) {
 	typeMeta := unversioned.TypeMeta{}
 	if err := json.Unmarshal(data, &typeMeta); err != nil {
-		return err
+		return nil, err
 	}
-	if typeMeta.Kind != t.kind {
-		return fmt.Errorf("unexpected kind: %s, expected %s", typeMeta.Kind, t.kind)
+	switch typeMeta.Kind {
+	case t.kind:
```

Comment:
>  this code path should never get called for any object that isn't under a third party group.

Right, but I think it's more robust if we check the group here. This won't block the PR, but I'd like to get your confirmation that there is nothing preventing us from making t.kind a GroupKind in the future.